cloud dm: add seoul region (#11979)
pingcap/docs: tidb-cloud/tidb-cloud-billing-dm.md

## Price

> For more information about the supported regions and the price of TiDB Cloud for each Data Migration RCU, see [Data Migration Cost](https://www.pingcap.com/tidb-cloud-pricing-details/#dm-cost).

Reviewer: ```suggestion
To learn about the supported regions and the price of TiDB Cloud for each Data Migration RCU, see [Data Migration Cost](https://www.pingcap.com/tidb-cloud-pricing-details/#dm-cost).
```
